multichain-testing/test/auto-stake-it.test.ts

```diff
@@ -0,0 +1,238 @@
+      throw Error(`failed to ibc transfer funds to ${chainName}`);
+    }
+    const { events: _events, ...txRest } = txRes;
+    console.log(txRest);
+    t.is(txRes.code, 0, `Transaction succeeded`);
+    t.log(`Funds transferred to ${agoricAddr}`);
+    return {
+      client,
+      address,
+      wallet,
+    };
+  };
+};
+
+const autoStakeItScenario = test.macro({
+  title: (_, chainName: string) => `auto-stake-it on ${chainName}`,
+  exec: async (t, chainName: string) => {
+    const {
+      wallets,
+      makeQueryTool,
+      provisionSmartWallet,
+      retryUntilCondition,
+    } = t.context;
+
+    const fundAndTransfer = makeFundAndTransfer(t);
+
+    // 1. Send initial tokens so denom is available (debatably necessary, but
+    // allows us to trace the denom until we have ibc denoms in chainInfo)
+    const agAdminAddr = wallets['agoricAdmin'];
+    console.log('Sending tokens to', agAdminAddr, `from ${chainName}`);
+    await fundAndTransfer(chainName, agAdminAddr);
+
+    // 2. Find 'stakingDenom' denom on agoric
+    const agoricConns = chainInfo['agoric'].connections as Record<
+      string,
+      IBCConnectionInfo
+    >;
+    const remoteChainInfo = (chainInfo as Record<string, CosmosChainInfo>)[
+      chainName
+    ];
+    // const remoteChainId = remoteChainInfo.chain.chain_id;
+    // const agoricToRemoteConn = agoricConns[remoteChainId];
+    const { portId, channelId } =
+      agoricConns[remoteChainInfo.chainId].transferChannel;
+    const agoricQueryClient = makeQueryClient(
+      useChain('agoric').getRestEndpoint(),
+    );
+    const stakingDenom = remoteChainInfo?.stakingTokens?.[0].denom;
+    if (!stakingDenom) throw Error(`staking denom found for ${chainName}`);
+    const { hash } = await retryUntilCondition(
+      () =>
+        agoricQueryClient.queryDenom(`/${portId}/${channelId}`, stakingDenom),
+      denomTrace => !!denomTrace.hash,
+      `local denom hash for ${stakingDenom} found`,
+    );
+    t.log(`found ibc denom hash for ${stakingDenom}:`, hash);
+
+    // 3. Find a remoteChain validator to delegate to
+    const remoteQueryClient = makeQueryClient(
+      useChain(chainName).getRestEndpoint(),
+    );
+    const { validators } = await remoteQueryClient.queryValidators();
+    const validatorAddress = validators[0]?.operator_address;
+    t.truthy(
+      validatorAddress,
+      `found a validator on ${chainName} to delegate to`,
+    );
+    t.log(
+      { validatorAddress },
+      `found a validator on ${chainName} to delegate to`,
+    );
+
+    // 4. Send an Offer to make the accounts and set up the transfer tap
+    const agoricUserAddr = wallets[accounts[chainName]];
+    const wdUser = await provisionSmartWallet(agoricUserAddr, {
+      BLD: 100n,
+      IST: 100n,
+    });
+    const doOffer = makeDoOffer(wdUser);
+    t.log(`${chainName} makeAccount offer`);
+    const offerId = `${chainName}-makeAccountsInvitation-${Date.now()}`;
+
+    await doOffer({
```

Comment:
this abstraction doesn't seem to carry its weight.
```suggestion
    await wallet.offers.executeOffer({
```

Reply:
It calls into `seatLike`, which has a bit of logic to ensure we get payouts. Seems like a bit more than this to inline.